x-pack/osquerybeat/ext/osquery-extension/pkg/tables/elastic_file_analysis.go

```diff
@@ -0,0 +1,125 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build darwin
+// +build darwin
+
+package tables
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/elastic/beats/v7/x-pack/osquerybeat/internal/command"
+	"github.com/osquery/osquery-go/plugin/table"
+)
+
+func ExecuteStderr(ctx context.Context, name string, arg ...string) (out string, err error) {
+	cmd := exec.CommandContext(ctx, name, arg...)
+	var stderr bytes.Buffer
+	cmd.Stderr = &stderr
+
+	err = cmd.Run()
+	if err != nil {
+		return stderr.String(), err
+	}
+
+	return stderr.String(), nil
+}
+
+func FileAnalysisColumns() []table.ColumnDefinition {
+	return []table.ColumnDefinition{
+		table.TextColumn("path"),
+		table.TextColumn("mode"),
+		table.BigIntColumn("uid"),
+		table.BigIntColumn("gid"),
+		table.TextColumn("size"),
+		table.TextColumn("mtime"),
+		table.TextColumn("file_type"),
+		table.TextColumn("code_sign"),
+		table.TextColumn("dependencies"),
+		table.TextColumn("symbols"),
+		table.TextColumn("strings"),
+	}
+}
+
+func GetFileAnalysisGenerateFunc() table.GenerateFunc {
+	return func(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+		var results []map[string]string
+
+		pathConstraint, exists := queryContext.Constraints["path"]
+		if !exists {
+			return results, nil
+		}
+
+		var path *string
+		for _, constraint := range pathConstraint.Constraints {
+			constraintCopy := constraint
+			path = &constraintCopy.Expression
+			break
+		}
+
+		if path == nil {
+			return results, nil
+		}
+
+		// Validate and sanitize the input path
+		stat, err := os.Stat(*path)
+		if err != nil || !stat.Mode().IsRegular() {
+			return results, fmt.Errorf("invalid path: %s", *path)
```

Comment:
might be worth considering two separate cases:
1. if err != nil, return wrapped the error, check ```fmt.Errorf```
2. if !IsRegular() return "invalid path........" or "this not a file....." error